test/jsx.jsx

```diff
@@ -0,0 +1,102 @@
+'use strict'
+
+var test = require('tape')
+var u = require('unist-builder')
+var x = require('..')
+
+test('name', function (t) {
+  t.deepEqual(<a />, x('a'), 'should support a self-closing element')
+
+  t.deepEqual(<a>b</a>, x('a', 'b'), 'should support a value as a child')
+
+  var A = 'a'
+
+  t.deepEqual(<A />, x(A), 'should support an uppercase tag name')
+
+  t.deepEqual(
+    <a>{1 + 1}</a>,
+    x('a', '2'),
+    'should support expressions as children'
+  )
+
+  t.deepEqual(<></>, u('root', []), 'should support a fragment')
+
+  t.deepEqual(
+    <>a</>,
+    u('root', [u('text', 'a')]),
+    'should support a fragment with text'
+  )
+
+  t.deepEqual(
+    <>
+      <a />
+    </>,
+    u('root', [x('a')]),
+    'should support a fragment with an element'
+  )
+
+  t.deepEqual(
+    <>{-1}</>,
+    u('root', [u('text', '-1')]),
+    'should support a fragment with an expression'
+  )
+
+  var com = {acme: {a: 'A', b: 'B'}}
+
+  t.deepEqual(
+    <com.acme.a />,
+    x(com.acme.a),
+    'should support members as names (`a.b`)'
+  )
+
+  t.deepEqual(
+    <a b />,
+    x('a', {b: 'true'}),
+    'should support a boolean attribute'
+  )
+
+  t.deepEqual(
+    <a b="" />,
+    x('a', {b: ''}),
+    'should support a double quoted attribute'
+  )
+
+  t.deepEqual(
+    <a b='"' />,
+    x('a', {b: '"'}),
+    'should support a single quoted attribute'
+  )
+
+  t.deepEqual(
+    <a b={1 + 1} />,
+    x('a', {b: '2'}),
+    'should support expression value attributes'
+  )
+
+  var props = {a: 1, b: 2}
+
+  t.deepEqual(
+    <a {...props} />,
+    x('a', props),
+    'should support expression spread attributes'
+  )
+
+  t.deepEqual(
+    <a>
+      <b />c<d>e</d>
+      {1 + 1}
+    </a>,
+    x('a', [x('b'), 'c', x('d', 'e'), '2']),
+    'should support text, elements, and expressions in jsx'
+  )
+
+  t.deepEqual(
+    <a>
+      <>{1}</>
+    </a>,
+    x('a', '1'),
```

Comment:
based off this

```jsx
<dl>
  {Object.entries({a: 'a', b: 'b'}).map(([key, value]) => <><dt>{key}</dt><dd>{value}</dd><>)}
</dl>
```

would resolve as:

```html
<dl>
  <dt>a</dt>
  <dd>a</dd>
  <dt>b</dt>
  <dd>b</dd>
</dl>
```

is that accurate?

Reply:
That’s the idea! I’ll add it as a test!